build.sbt

```diff
@@ -1,92 +1,82 @@
+lazy val defaultScalaVer = sparkBranch match {
   case "3.5" => "2.12.18"
   case _ => throw new IllegalArgumentException(s"Unsupported Spark version: $sparkVer.")
 }
-val scalaVer = sys.props.getOrElse("scala.version", defaultScalaVer)
-val defaultScalaTestVer = scalaVer match {
+lazy val scalaVer = sys.props.getOrElse("scala.version", defaultScalaVer)
+lazy val defaultScalaTestVer = scalaVer match {
   case s if s.startsWith("2.12") || s.startsWith("2.13") => "3.0.8"
 }
 
-sparkVersion := sparkVer
-
-scalaVersion := scalaVer
-
-name := "graphframes"
-
-spName := "graphframes/graphframes"
-
-organization := "org.graphframes"
-
-version := (version in ThisBuild).value + s"-spark$sparkBranch"
-
-isSnapshot := version.value.contains("SNAPSHOT")
-
-// All Spark Packages need a license
-licenses := Seq("Apache-2.0" -> url("http://opensource.org/licenses/Apache-2.0"))
-
-spAppendScalaVersion := true
-
-// Add Spark components this package depends on, e.g, "mllib", ....
-sparkComponents ++= Seq("graphx", "sql", "mllib")
-
-// uncomment and change the value below to change the directory where your zip artifact will be created
-// spDistDirectory := target.value
-
-// add any Spark Package dependencies using spDependencies.
-// e.g. spDependencies += "databricks/spark-avro:0.1"
-
-libraryDependencies += "org.slf4j" % "slf4j-api" % "1.7.16"
-
-libraryDependencies += "org.scalatest" %% "scalatest" % defaultScalaTestVer % "test"
-
-libraryDependencies += "com.github.zafarkhaja" % "java-semver" % "0.9.0" % "test" // MIT license
-
-parallelExecution := false
-
-scalacOptions ++= Seq("-deprecation", "-feature")
-
-scalacOptions in (Compile, doc) ++= Seq(
-  "-groups",
-  "-implicits",
-  "-skip-packages", Seq("org.apache.spark").mkString(":"))
-
-scalacOptions in (Test, doc) ++= Seq("-groups", "-implicits")
-
-// This fixes a class loader problem with scala.Tuple2 class, scala-2.11, Spark 2.x
-fork in Test := true
-
-// This and the next line fix a problem with forked run: https://github.com/scalatest/scalatest/issues/770
-javaOptions in Test ++= Seq(
-  "-XX:+IgnoreUnrecognizedVMOptions",
-  "-Xmx2048m",
-  "-XX:ReservedCodeCacheSize=384m",
-  "-XX:MaxMetaspaceSize=384m",
-  "--add-opens=java.base/sun.nio.ch=ALL-UNNAMED",
-  "--add-opens=java.base/java.lang=ALL-UNNAMED"
-)
-
-concurrentRestrictions in Global := Seq(
-  Tags.limitAll(1))
-
-autoAPIMappings := true
-
-coverageHighlighting := false
-
-// We only use sbt-release to update version numbers.
-releaseProcess := Seq[ReleaseStep](
-  inquireVersions,
-  setReleaseVersion,
-  commitReleaseVersion,
-  tagRelease,
-  setNextVersion,
-  commitNextVersion
-)
+ThisBuild / version := {
+  val baseVersion = (ThisBuild / version).value
+  s"${baseVersion}-spark${sparkBranch}"
+}
 
-credentials += Credentials(Path.userHome / ".ivy2" / ".sbtcredentials")
+ThisBuild / scalaVersion := scalaVer
+ThisBuild / organization := "org.graphframes"
+
+lazy val root = (project in file("."))
```

Comment:
I need this for adding Spark Connect support: connect 100% should be a separate subproject and it will be much easier to write `(project in file("connect")).dependsOn(root)` to inherit most of the settings and use `graphframes` (core) as a dependency. As a bonus it is more modern sbt syntax that is recommended by their documentation (instead of implicit project in root)